src/types/basic.rs

```diff
@@ -694,6 +694,18 @@ pub struct UpdateKeysThreshold {
     pub(crate) threshold: u16,
 }
 
+impl TryFrom<u16> for UpdateKeysThreshold {
+    type Error = &'static str;
```

Comment:
I would prefer a proper error type here.

Reply:
Yes. It's just another unit type though :)

But will add.

Reply:
Added.